services/venice-router/src/main/java/com/linkedin/venice/router/throttle/ReadRequestThrottler.java

```diff
@@ -253,6 +260,9 @@ public void handleRouterCountChanged(int newRouterCount) {
 
   @Override
   public void onExternalViewChange(PartitionAssignment partitionAssignment) {
+    if (!perStorageNodeThrottlerEnabled) {
```

Comment:
So if `perStorageNodeThrottlerEnabled` is disabled, will this callback ever be triggered?
This line might be harmless, but confusing.

Reply:
yes, ideally it wont be triggered, its just a sanity check. goal is to remove all the code altogether when we can disable the config